internal/engine/compiler/impl_amd64.go

```diff
@@ -5013,3 +5082,43 @@ func (c *amd64Compiler) compileEnsureOnRegister(loc *runtimeValueLocation) (err
 	}
 	return
 }
+
+// compileMaybeSwapRegisters swaps two registers if they're not equal.
+func (c *amd64Compiler) compileMaybeSwapRegisters(reg1, reg2 asm.Register) {
+	if reg1 != reg2 {
+		c.assembler.CompileRegisterToRegister(amd64.XCHGQ, reg1, reg2)
+	}
+}
+
+// compilePreventCrossedTargetRegisters ensures that neither runtimeValueLocations target register
```

Comment:
this sentence is hard to understand. Could you elaborate more on what `runtimeValueLocations target register` means and please add the example here.

Also,use exactly the same name as in function params `locs` and `targets` in the documentation. That's how we reduce the overhead to understand how the function performs without looking at the function body. Please follow the same rule for all of the your implementation.